pf4j/src/main/java/org/pf4j/AbstractPluginManager.java

```diff
@@ -820,18 +819,31 @@ protected synchronized void firePluginStateEvent(PluginStateEvent event) {
     }
 
     protected PluginWrapper loadPluginFromPath(Path pluginPath) throws PluginException {
-        // test for plugin duplication
+        // Test for plugin path duplication
         String pluginId = idForPath(pluginPath);
         if (pluginId != null) {
             throw new PluginAlreadyLoadedException(pluginId, pluginPath);
         }
 
-        // retrieves the plugin descriptor
+        // Retrieve and validate the plugin descriptor
         PluginDescriptorFinder pluginDescriptorFinder = getPluginDescriptorFinder();
         log.debug("Use '{}' to find plugins descriptors", pluginDescriptorFinder);
         log.debug("Finding plugin descriptor for plugin '{}'", pluginPath);
         PluginDescriptor pluginDescriptor = pluginDescriptorFinder.find(pluginPath);
         validatePluginDescriptor(pluginDescriptor);
+
+        // Check there are no loaded plugins with the retrieved id
+        pluginId = pluginDescriptor.getPluginId();
+        if (plugins.containsKey(pluginId)) {
+            PluginWrapper loadedPlugin = getPlugin(pluginId);
+            throw new PluginException(String.format("There is an already loaded plugin (%s) "
```

Comment:
Only a hint. You can write a little bit small code using [PluginException(String message, Object... args)](https://github.com/pf4j/pf4j/blob/master/pf4j/src/main/java/org/pf4j/PluginException.java#L48). The format is similar with slf4j:
```java
throw new PluginException(("There is an already loaded plugin ({}) "
                    + "with the same id ({}) as the plugin at path '{}'. Simultaneous loading "
                    + "of plugins with the same PluginId is not currently supported.\n"
                    + "As a workaround you may include PluginVersion and PluginProvider "
                    + "in PluginId.",
                loadedPlugin, pluginId, pluginPath);  
```

It's my fault that `PluginException` doesn't come with a javadoc for its methods.

Reply:
Thanks, that's definitely more convenient!

I first tried to use `PluginAlreadyLoadedException`, but it has some required fields (pluginId and pluginPath) that might be ambiguous to set in this case.